### PR TITLE
MBSA-484 Add Dockerfile for Emcee worker

### PR DESCRIPTION
**Pull request by RuslanMingaliev**

### CODE REVIEW

File: ci/docker/emcee-worker/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,24 @@
+ARG DOCKER_REGISTRY
```
Comment (sergioboi) [inline ci/docker/emcee-worker/Dockerfile:1]: What is this?

File: ci/docker/emcee-worker/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+RUN mkdir /app
+COPY . /app
+WORKDIR /app
+
+RUN ./gradlew subprojects:emcee:worker:build \
+    -PartifactoryUrl=${ARTIFACTORY_URL} \
+    -Pcom.avito.android.tools.buildCache.remote.url=${GRADLE_GITHUB_BUILD_CACHE_URL}
+
+RUN ls subprojects/emcee/worker/build/libs
```
Comment (sergioboi) [inline ci/docker/emcee-worker/Dockerfile]: why do we need this line?

**Review by sergioboi — APPROVED** [review]

File: ci/docker/emcee-worker/Dockerfile (hunk 1 above)
Comment (RuslanMingaliev) [reply ci/docker/emcee-worker/Dockerfile:1]: Our internal storage for docker images

File: ci/docker/emcee-worker/Dockerfile (hunk 2 above)
Comment (RuslanMingaliev) [reply ci/docker/emcee-worker/Dockerfile]: Forgot to remove that, thx

File: ci/docker/emcee-worker/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+ARG ARTIFACTORY_URL
+
+RUN mkdir /app
+COPY . /app
+WORKDIR /app
+
+RUN ./gradlew subprojects:emcee:worker:build \
+    -PartifactoryUrl=${ARTIFACTORY_URL} \
+    -Pcom.avito.android.tools.buildCache.remote.url=unused \
+    --no-build-cache
```
Comment (sergioboi) [inline ci/docker/emcee-worker/Dockerfile:14]: why no buildCache?

File: ci/docker/emcee-worker/config.json [hunk 4]
```diff
@@ -0,0 +1,5 @@
+{
+    "workerPort": 80,
+    "queueUrl": "http://emcee-queue-service:41000",
+    "androidSdkPath": "/Users/john/androidSdk"
```
Comment (sergioboi) [inline ci/docker/emcee-worker/config.json:4]: This is path in docker file?

File: subprojects/emcee/worker/src/main/kotlin/com/avito/emcee/worker/internal/identifier/HostnameWorkerIdProvider.kt [hunk 5]
```diff
@@ -0,0 +1,10 @@
+package com.avito.emcee.worker.internal.identifier
+
+import java.net.InetAddress
+
+internal class HostnameWorkerIdProvider : WorkerIdProvider {
+
+    override fun provide(): String {
+        return InetAddress.getLocalHost().hostName
```
Comment (sergioboi) [inline subprojects/emcee/worker/src/main/kotlin/com/avito/emcee/worker/internal/identifier/HostnameWorkerIdProvider.kt:8]: Are you sure that host name will be different through the workers?

File: subprojects/emcee/worker/src/main/kotlin/com/avito/emcee/worker/internal/identifier/WorkerIdProvider.kt [hunk 6]
```diff
@@ -0,0 +1,5 @@
+package com.avito.emcee.worker.internal.identifier
+
+internal interface WorkerIdProvider {
+    fun provide(): String
```
Comment (sergioboi) [inline subprojects/emcee/worker/src/main/kotlin/com/avito/emcee/worker/internal/identifier/WorkerIdProvider.kt:4]: Let's use inline/value class WorkerId(id:String)

File: subprojects/emcee/worker/src/main/kotlin/com/avito/emcee/worker/internal/rest/handler/HealthCheckRequestHandler.kt [hunk 7]
```diff
@@ -0,0 +1,10 @@
+package com.avito.emcee.worker.internal.rest.handler
+
+import io.ktor.http.HttpMethod
+
+// A simple way to check if worker is alive using HTTP request
+internal object HealthCheckRequestHandler : RequestHandler<String>(
+    method = HttpMethod.Get,
+    path = "/healthCheck",
+    response = { "I'm alive" }
```
Comment (sergioboi) [inline subprojects/emcee/worker/src/main/kotlin/com/avito/emcee/worker/internal/rest/handler/HealthCheckRequestHandler.kt:9]: I think better to send 0/1. Less data

File: ci/docker/emcee-worker/config.json (hunk 4 above)
Comment (RuslanMingaliev) [reply ci/docker/emcee-worker/config.json:4]: It's a path in the docker container

File: subprojects/emcee/worker/src/main/kotlin/com/avito/emcee/worker/internal/identifier/HostnameWorkerIdProvider.kt (hunk 5 above)
Comment (RuslanMingaliev) [reply subprojects/emcee/worker/src/main/kotlin/com/avito/emcee/worker/internal/identifier/HostnameWorkerIdProvider.kt:8]: yes, I checked

File: subprojects/emcee/worker/src/main/kotlin/com/avito/emcee/worker/internal/rest/handler/HealthCheckRequestHandler.kt (hunk 7 above)
Comment (RuslanMingaliev) [reply subprojects/emcee/worker/src/main/kotlin/com/avito/emcee/worker/internal/rest/handler/HealthCheckRequestHandler.kt:9]: Agree

File: subprojects/emcee/worker/src/main/kotlin/com/avito/emcee/worker/internal/identifier/WorkerIdProvider.kt (hunk 6 above)
Comment (RuslanMingaliev) [reply subprojects/emcee/worker/src/main/kotlin/com/avito/emcee/worker/internal/identifier/WorkerIdProvider.kt:4]: Okay

File: subprojects/emcee/worker/src/main/kotlin/com/avito/emcee/worker/internal/identifier/HostnameWorkerIdProvider.kt (hunk 5 above)
Comment (sergioboi) [reply subprojects/emcee/worker/src/main/kotlin/com/avito/emcee/worker/internal/identifier/HostnameWorkerIdProvider.kt:8]: I hope it wont break. Now it looks fragile. Maybe we could make a fail-fast check for that some how

**Pull request merged**